src/interface.jl

```diff
@@ -209,11 +209,19 @@ valid_function_pointer(@nospecialize(job::CompilerJob), ptr::Ptr{Cvoid}) = false
 
 # the codeinfo cache to use
 function ci_cache(@nospecialize(job::CompilerJob))
-    lock(GLOBAL_CI_CACHES_LOCK) do
-        cache = get!(GLOBAL_CI_CACHES, job.config) do
-            CodeCache()
+    @static if isdefined(Core.Compiler, :cache_owner)
+        # CompilerConfig contains
+        # - name
+        # - params (Enzyme puts info here...)
+        # unecessary bifurcation
+        return Core.Compiler.InternalCodeCache(job.config) 
+    else
+        lock(GLOBAL_CI_CACHES_LOCK) do
+            cache = get!(GLOBAL_CI_CACHES, job.config) do
```

Comment:
@maleadt Already now I feel like we are bifurcating the caches to much.

I think this means Enzyme creates a fresh inference cache per top-level request,
due to us using `EnzymeCompilerParams` to stash information.

https://github.com/EnzymeAD/Enzyme.jl/blob/3f641088fa39cf0cfb00ffd9e4b8c8200564dbfc/src/compiler.jl#L2477

cc: @wsmoses